### PR TITLE
ASoC: Intel: tgl: add tgl-rt1308 machine driver

### PR DESCRIPTION
**Pull request by reddymur**

### CODE REVIEW

**Comment by sofci** [comment]
Can one of the admins verify this patch?
> reply **test&nbsp;this&nbsp;please** to run this test once

**Review by paulstelian97 — CHANGES_REQUESTED** [review]
Too much commented out code, is there any good reason to have it?

File: sound/soc/intel/boards/tgl_rt1308.c
Comment (paulstelian97) [inline sound/soc/intel/boards/tgl_rt1308.c:187]: Is Xiuli the one who made this file specifically? Or is this just a copy-paste issue?

Comment (reddymur) [reply sound/soc/intel/boards/tgl_rt1308.c:187]: Xiuli - had made this file in yocto linux v5.4. But now for IoTG requirements we are porting it to yocto linux v5.10.

**Comment by reddymur** [comment]
> Too much commented out code, is there any good reason to have it?

**Comment by paulstelian97** [comment]
Did you close it on purpose? Or shall we reopen it?

**Comment by reddymur** [comment]
paulstelian97  - it was not closed on purpose, happened by mistake. pls review.

**Review by paulstelian97 — APPROVED** [review]
I guess this looks fine, I'll wait for other maintainers who can check this better to review.

File: sound/soc/intel/boards/tgl_rt1308.c
Comment (kv2019i) [inline sound/soc/intel/boards/tgl_rt1308.c:3]: Copyright can be updated.

File: sound/soc/intel/boards/Kconfig
Comment (kv2019i) [inline sound/soc/intel/boards/Kconfig:524]: Replace L524 with:
depends on SND_HDA_CODEC_HDMI && SND_SOC_SOF_HDA_AUDIO_CODEC

File: sound/soc/intel/boards/tgl_rt1308.c
Comment (kv2019i) [inline sound/soc/intel/boards/tgl_rt1308.c:23]: This L23 can be dropped.

File: sound/soc/intel/boards/tgl_rt1308.c
Comment (kv2019i) [inline sound/soc/intel/boards/tgl_rt1308.c:34]: This too L29-34

File: sound/soc/intel/boards/tgl_rt1308.c
Comment (kv2019i) [inline sound/soc/intel/boards/tgl_rt1308.c:163]: The HDMI support seems a bit partial. We don't support HDAC_HDMI with new mach drivers, so I recommend either dropping it (like this L162-163), or adding HDMI support like e.g. sof_pcm512x.c are other more recent machine drivers.

**Review by kv2019i — CHANGES_REQUESTED** [review]
Thank you @reddymur ! Some feedback inline

File: sound/soc/intel/boards/tgl_rt1308.c
Comment (plbossart) [inline sound/soc/intel/boards/tgl_rt1308.c:27]: why do you declare this when there are no dailinks with HDMI?

Is this really intentional to only provide support for DMIC + amplifier?

File: sound/soc/intel/boards/tgl_rt1308.c
Comment (plbossart) [inline sound/soc/intel/boards/tgl_rt1308.c:189]: GPL

File: sound/soc/intel/boards/tgl_rt1308.c
Comment (plbossart) [inline sound/soc/intel/boards/tgl_rt1308.c:187]: need to find a different author.

File: sound/soc/intel/boards/tgl_rt1308.c
Comment (plbossart) [inline sound/soc/intel/boards/tgl_rt1308.c:163]: not needed if you don't care about HDMI

File: sound/soc/intel/boards/tgl_rt1308.c
Comment (plbossart) [inline sound/soc/intel/boards/tgl_rt1308.c:113]: remove .nonatomic, this is not right

**Review by plbossart — CHANGES_REQUESTED** [review]

**Comment by plbossart** [comment]
@reddymur @kv2019i counter-proposal in https://github.com/thesofproject/linux/pull/3000

**Comment by CBala21** [comment]
> @reddymur @kv2019i counter-proposal in #3000

> @kv2019i  We don't have issue till can achieve the same I2s amplifier +DMIC with this driver. Also require the provision to add dummy codec for slave mode configuration.

**Comment by plbossart** [comment]
> > @reddymur @kv2019i counter-proposal in #3000
> 
> > @kv2019i  We don't have issue till can achieve the same I2s amplifier +DMIC with this driver. Also require the provision to add dummy codec for slave mode configuration.

The slave mode can be handled with the existing 'nocodec' machine driver and a topology change. This is a different problem.
If you want RT1308 on one I2S and dummy codecs on other links that's not supported.

**Comment by CBala21** [comment]
> > > @reddymur @kv2019i counter-proposal in #3000
> > 
> > 
> > > @kv2019i  We don't have issue till can achieve the same I2s amplifier +DMIC with this driver. Also require the provision to add dummy codec for slave mode configuration.
> 
> The slave mode can be handled with the existing 'nocodec' machine driver and a topology change. This is a different problem.
> If you want RT1308 on one I2S and dummy codecs on other links that's not supported.

@plbossart : No, the requirement is RT1308 playback with SSP2 and HDMI receiver in SSP0/1 with slave mode configuration. its exactly like ADL-BT offload use case. kai has back ground of this requirement and indicated by mail.
I guess If we can't meet this requirement with SDW machine driver then separate I2S machine driver is the approach to be followed. please check and suggest.

**Comment by plbossart** [comment]
> @plbossart : No, the requirement is RT1308 playback with SSP2 and HDMI receiver in SSP0/1 with slave mode configuration. its exactly like ADL-BT offload use case. kai has back ground of this requirement and indicated by mail.
> I guess If we can't meet this requirement with SDW machine driver then separate I2S machine driver is the approach to be followed. please check and suggest.

D'oh!

Your initial ask was to upstream the same driver that was provided earlier by Xiuli Pan. 
Now you are asking for something different. What you had for 5.4 NEVER supported the slave mode and dummy codecs. It's not even in the proposal you made in https://github.com/thesofproject/linux/pull/2993, so your proposal does not even comply with your own requirements.

Please huddle with your team and clarify what is it that you want.

**Comment by plbossart** [comment]
This was a spectacular waste of my time.

**Comment by CBala21** [comment]
> > @plbossart : No, the requirement is RT1308 playback with SSP2 and HDMI receiver in SSP0/1 with slave mode configuration. its exactly like ADL-BT offload use case. kai has back ground of this requirement and indicated by mail.
> > I guess If we can't meet this requirement with SDW machine driver then separate I2S machine driver is the approach to be followed. please check and suggest.
> 
> D'oh!
> 
> Your initial ask was to upstream the same driver that was provided earlier by Xiuli Pan.
> Now you are asking for something different. What you had for 5.4 NEVER supported the slave mode and dummy codecs. It's not even in the proposal you made in #2993, so your proposal does not even comply with your own requirements.
> 
> Please huddle with your team and clarify what is it that you want.

 @plbossart :  Hi,  With out base RT1308 driver we cant implement and meet customer new requirement. that's the reason we requested to upstream the RT1308 driver first. then we will implement the new requirement top on base machine driver and can be up streamed it later. Also this RT1308 driver required first to meet platform POR.  if we can meet platform POR ( I2S based RT 1308 ) and customer new requirement(HDMI receiver in I2S slave) by using SDW machine driver then we are ok to follow your approach. 
Apologies for the confusion.

**Pull request closed without merging**
